website/docs/concepts/why_immutability.mdx

```diff
@@ -0,0 +1,113 @@
+---
+title: Why Immutability
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+## What is Immutability?
+
+Immutability is essentially when all fields of an `Object` are final.
+They are set exactly once upon construction.
+
+Immutability is desireable for many different reasons
+
+- Value equality rather than reference equality
+  - A copyWith method helps with reducing verbosity when creating a new object with just a few things changed
+- Local reasoning about a piece of code
+  - A far distant piece of code can't obtain a reference and change the object from underneath you
+- Easier to reason about for asynchronous and parallel tasks
+  - Other code can't mutate your object in between operations
+- Safety of APIs
+  - What you pass into a method cannot be changed by the callee / caller
+  - Flutter uses this for Themes, since an override of a theme parameter in one part of an app shouldn't affect another part
+
+## Best Practices
+
+You can use any package you want to create immutable state.
+
+For immutable objects:
+
+- [package:freezed](https://pub.dev/packages/freezed)
+- [package:built_value](https://pub.dev/packages/built_value)
+
+For immutable collections (Map, Set, List):
+
+- [package:fast_immutable_collections](https://pub.dev/packages/fast_immutable_collections)
+- [package:built_collection](https://pub.dev/packages/built_collection)
+- [package:kt_dart](https://pub.dev/packages/kt_dart)
+- [package:dartz](https://pub.dev/packages/dartz)
+
+It is highly recommended to use `freezed`,
+since it has several nice additions beyond just making immutable objects including:
+
+- A generated copyWith method
+- Deep copy (copyWith on nested freezed objects)
+- Union types
+- Union mapping functions
```

Comment:
I think it'd be helpful to expand on those two, especially "union types" (likely outside of "why Freezed")

That's probably the most important part of Immutability/StateNotifier vs using ChangeNotifier, to allow making impossible states impossible